common/service.go

```diff
@@ -6,101 +6,90 @@ import (
 	"github.com/tendermint/tmlibs/log"
 )
 
-type Service interface {
-	Start() (bool, error)
+// ServiceCore is a service that is implemented by a user.
+type ServiceCore interface {
 	OnStart() error
+	OnStop() error
+}
 
-	Stop() bool
-	OnStop()
-
-	Reset() (bool, error)
-	OnReset() error
-
+// Service represents a way to start, stop and get the status of some service. BaseService is the
+// default implementation and should be used by most users.
+// SetServiceCore allows a user to set a ServiceCore. Users should implement their logic using
+// ServiceCore.
+type Service interface {
+	Start() error
+	Stop() error
 	IsRunning() bool
-
-	String() string
-
+	SetServiceCore(ServiceCore)
```

Comment:
I like the new usage of `ServiceCore`. However, shouldn't we also make ServiceCore implement the String() method if Service implicitly invokes OnStart() and OnStop() of the ServiceCore?

Reply:
The String is just the name passed into NewBaseService